src/storage/compression/roaring/common.cpp

```diff
@@ -198,9 +198,16 @@ unique_ptr<CompressionState> RoaringInitCompression(ColumnDataCheckpointData &ch
 	return make_uniq<RoaringCompressState>(checkpoint_data, std::move(state));
 }
 
+template <PhysicalType TYPE>
 void RoaringCompress(CompressionState &state_p, Vector &scan_vector, idx_t count) {
 	auto &state = state_p.Cast<RoaringCompressState>();
-	state.Compress(scan_vector, count);
+	if (TYPE == PhysicalType::BOOL) {
+		auto data_ptr = scan_vector.GetData();
+		for (idx_t i = 0; i < count; i++) {
+			state.current_segment->stats.statistics.UpdateNumericStats<bool>(data_ptr[i]);
```

Comment:
I wonder if we can somehow push this into the bitpacking step, since we're already looping over the vector in its entirety there as well